File: MagneticSensor.java
public class MagneticSensor extends Sensor {
    public MagneticSensor(int z){
        super(z);
        view= new MagneticSensorView();
    }
    public void setSensorOpen() {
        setState(SwitchState.OPEN);
        view.setOpenView();
    }
    public void setSensorClose() {
        setState(SwitchState.CLOSE);
        view.setCloseView();
    }
    public MagneticSensorView getView(){ return view;}
    private final MagneticSensorView view;
}

File: magnetic_sensor.cpp
#include "magnetic_sensor.h"

MagneticSensor::MagneticSensor(int z): Sensor(z,true) {
}
void MagneticSensor::setSensorOpen() {
    setClose(false);
}
void MagneticSensor::setSensorClose() {
    setClose(true);
}
